nextjs-13/src/app/newsletter/[id]/page.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { ClipLoader } from 'react-spinners';
+
+import { useParams, redirect } from 'next/navigation';
+
+import NewsletterDetail from '@components/NewsletterDetail';
+import requestManager from 'lib/request/manager';
+
+const ViewNewsletter = () => {
+  const [newsletter, setNewsletter] = useState();
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const params = useParams();
+  const id = params.id;
+
+  useEffect(() => {
+    const fetchRecord = async () => {
+      setIsLoading(true);
+
+      try {
+        const response = await requestManager('GET', `v1/newsletter/${id}`);
+
+        setIsLoading(false);
+        setNewsletter(response.record);
+      } catch (error) {
+        setHasError(true);
+      }
+    };
+
+    fetchRecord();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
```

Comment:
I use another `useEffect` here because if I use `redirect` at line 28 instead, it raises [this error](https://github.com/vercel/next.js/issues/42556). Seems like `redirect` can not be used in async function(`fetchRecord`)?